daphne_worker/src/durable/mod.rs

```diff
@@ -94,42 +108,61 @@ impl<'a> DurableConnector<'a> {
     ) -> Result<O> {
         let namespace = self.env.durable_object(durable_binding)?;
         let stub = namespace.id_from_string(&durable_id_hex)?.get_stub()?;
-        durable_request(stub, durable_path, Method::Post, Some(data))
+        self.durable_request(stub, durable_path, Method::Post, Some(data))
             .await
             .map_err(|error| {
                 Error::RustError(format!(
                     "DO {durable_binding}: post {durable_path}: {error}"
                 ))
             })
     }
-}
 
-async fn durable_request<I: Serialize, O: for<'a> Deserialize<'a>>(
-    durable_stub: Stub,
-    durable_path: &'static str,
-    method: Method,
-    data: Option<I>,
-) -> Result<O> {
-    let req = match (&method, data) {
-        (Method::Post, Some(data)) => Request::new_with_init(
-            &format!("https://fake-host{durable_path}"),
-            RequestInit::new().with_method(Method::Post).with_body(Some(
-                wasm_bindgen::JsValue::from_str(&serde_json::to_string(&data)?),
-            )),
-        )?,
-        (Method::Get, None) => Request::new_with_init(
-            &format!("https://fake-host{durable_path}"),
-            RequestInit::new().with_method(Method::Get),
-        )?,
-        _ => {
-            return Err(Error::RustError(format!(
-                "durable_request: Unrecognized method: {method:?}",
-            )));
-        }
-    };
+    async fn durable_request<I: Serialize, O: for<'a> Deserialize<'a>>(
+        &self,
+        durable_stub: Stub,
+        durable_path: &'static str,
+        method: Method,
+        data: Option<I>,
+    ) -> Result<O> {
+        let req = match (&method, data) {
+            (Method::Post, Some(data)) => Request::new_with_init(
+                &format!("https://fake-host{durable_path}"),
+                RequestInit::new().with_method(Method::Post).with_body(Some(
+                    wasm_bindgen::JsValue::from_str(&serde_json::to_string(&data)?),
+                )),
+            )?,
+            (Method::Get, None) => Request::new_with_init(
+                &format!("https://fake-host{durable_path}"),
+                RequestInit::new().with_method(Method::Get),
+            )?,
+            _ => {
+                return Err(Error::RustError(format!(
+                    "durable_request: Unrecognized method: {method:?}",
+                )));
+            }
+        };
 
-    let mut resp = durable_stub.fetch_with_request(req).await?;
-    resp.json().await
```

Comment:
Reviewer note: Just try the request a few times, waiting the same amount of time between each retry. This is pretty redumentary -- Ideally we'd want to do some sort of exponential scheme, like https://docs.rs/backoff/latest/backoff/index.html. However the backoff crate in particular doesn't seem to play well with miniflare (haven't tried in Workers). I think the issue is that it wants to sample the clock but the runtime doesn't have a callback configured.

Reply:
The standard "no frills" exponential backoff would be to set "let mut retry_interval = DEFAULT_RETRY_INTERVAL" right after the "let mut attempt = 1" and then do retry_interval *= 2 after the attempt += 1.

Reply:
Yeah that's what I was planning initially, but then I figured it would be good to add jitter as well. We may need to re-implement this at some point, but not now!